clients/src/main/java/org/apache/kafka/clients/consumer/Consumer.java

```diff
@@ -71,12 +71,12 @@
     /**
      * @see KafkaConsumer#commit(Map, CommitType)
      */
-    public void commit(Map<TopicPartition, Long> offsets, CommitType commitType);
+    public void commit(Map<TopicPartition, OffsetMetadata> offsets, CommitType commitType);
```

Comment:
@ewencp List might not be great, but Collection might work. The user could still use a Map, and then pass map.values() to commit. 

Also, that's a good point on mutability for the async case. I'd rather not allow that data to change while the commit is still pending, so maybe an internal copy is the best answer?